### PR TITLE
NetteExtension total separation

### LINKED ISSUE #4: Split NetteExtension to each component into separate bridges in backwards compatible way
- [x] database @hrach 
- [x] cache @dg
- [ ] http @xificurk 
- [ ] tracy @xificurk  
- [x] mail @xificurk  
- [x] application @dg 
- [x] routing @dg 
- [ ] security
- [ ] forms
- [x] latte @xificurk 
- [ ] container

### PR DESCRIPTION
**Pull request by dg**
Closes #4

### CODE REVIEW

**Comment by xificurk** [comment]
:+1: but my notes from nette/di#35 apply here as well.

**Comment by dg** [comment]
fixed mailer => mail

**Comment by dg** [comment]
BC break: 'Nette\Utils\SafeStream::register()' in initialize() was removed

**Comment by enumag** [comment]
@dg Is this intended for Nette 2.3 or Nette 2.4 / 3.0?

**Comment by dg** [comment]
2.3

File: src/Bootstrap/Configurator.php [hunk 1]
```diff
@@ -36,13 +36,19 @@ class Configurator extends Object
...
 		'extensions' => 'Nette\DI\Extensions\ExtensionsExtension',
 		'decorator' => 'Nette\DI\Extensions\DecoratorExtension',
-		'nette' => 'Nette\Bridges\Framework\NetteExtension',
 		'application' => array('Nette\Bridges\ApplicationDI\ApplicationExtension', array('%debugMode%')),
 		'routing' => array('Nette\Bridges\ApplicationDI\RoutingExtension', array('%debugMode%')),
 		'latte' => array('Nette\Bridges\ApplicationDI\LatteExtension', array('%tempDir%/cache/latte', '%debugMode%')),
 		'cache' => array('Nette\Bridges\CacheDI\CacheExtension', array('%tempDir%')),
 		'database' => 'Nette\Bridges\DatabaseDI\DatabaseExtension',
 		'mail' => 'Nette\Bridges\MailDI\MailExtension',
+		'DI' => array('Nette\DI\Extensions\DIExtension', array('%debugMode%')),
```
Comment (fprochazka) [inline src/Bootstrap/Configurator.php]: `di` ?

Comment (dg) [inline src/Bootstrap/Configurator.php]: asi jo…

File: composer.json [hunk 2]
```diff
@@ -35,6 +35,8 @@
 		"tracy/tracy": "~2.3"
 	},
 	"conflict": {
+		"nette/database": "<2.3",
+		"nette/di": "<2.3",
```
Comment (fprochazka) [inline composer.json]: why this constraint?

Comment (dg) [inline composer.json]: I don't know 8-)

**Pull request merged**